config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/loader/ServiceLoader.java

```diff
@@ -43,7 +45,9 @@ protected String getFileName() {
 
     @Override
     protected void putToBundle(Bundle bundle, @NotNull Map<String, Service> entitiesMap) {
-        bundle.putAllServices(entitiesMap);
+        Map<String, Service> parsedServices = new HashMap<>(entitiesMap.size());
+        entitiesMap.forEach((k, s) -> parsedServices.put(EncodeDecodeUtils.decodePath(k), s));
```

Comment:
If we do the decoding here, we will lose the information of service or folder names with `_¯`. Instead should we be decoding in the ServiceEntityBuilder?

Reply:
This is also done in the FolderAndPolicyLoader: com/ca/apim/gateway/cagatewayconfig/tasks/zip/loader/PolicyAndFolderLoader.java:71

When FolderLoaderUtils#createFolders the paths given should already be encoded.

Reply:
Actually you are right, I'll move all the decoding to the entity builders